src/components/Slider/Slider.tsx

```diff
@@ -0,0 +1,44 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { IconProps } from "@components/Icon/Icon";
+import { Dispatch, ReactElement, SetStateAction } from "react";
+import css from "./Slider.module.css";
+
+export interface IconItem {
+    icon: ReactElement<IconProps>;
+    name: string;
+}
+
+export interface SliderProps {
+    items: string[] | IconItem[];
+    activeItem: string;
+    onChange: Dispatch<SetStateAction<string>>;
+}
+
+const isIconItem = (item: string | IconItem): item is IconItem => (item as IconItem).icon !== undefined;
+
+export default function Slider({ items, activeItem: active, onChange }: SliderProps): ReactElement<SliderProps> {
+    return (
+        <div data-test-id="Slider" className={css.wrapper}>
+            {items.map((item, i) =>
+                isIconItem(item) ? (
+                    <div
+                        key={i}
```

Comment:
if the item has an `id`-prop use this instead of `i`. also using the array index is discouraged to use if there is another unique value you can use. (also see comment on SO https://stackoverflow.com/a/46735689 why using index should be avoided)